libs/logdiag/logdiag.go

```diff
@@ -0,0 +1,173 @@
+package logdiag
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"sync"
+
+	"github.com/databricks/cli/libs/cmdio"
+	"github.com/databricks/cli/libs/diag"
+)
+
+type keyType int
+
+const key = keyType(2)
+
+var Mu sync.Mutex
+
+type Value struct {
+	// How many diagnostics if each type were seen
+	Errors          int
+	Warnings        int
+	Recommendations int
+
+	// Only print diagnostics with this severity or lower (default: everything)
+	TargetSeverity diag.Severity
+
+	// Root to resolve location against
+	Root string
+
+	// If Collect is true, diagnostics are appended to Collected. Use SetCollected() to set.
+	Collect   bool
+	Collected []diag.Diagnostic
+}
+
+// IsSetup is test helper, do not use in production code
+func IsSetup(ctx context.Context) bool {
+	Mu.Lock()
+	defer Mu.Unlock()
+
+	_, ok := ctx.Value(key).(*Value)
+	return ok
+}
+
+func InitContext(ctx context.Context) context.Context {
```

Comment:
We have context getters and setters in `libs/cmdctx`. This can also be moved there to keep context interactions contained within that package.

Reply:
The common thread is that these are singleton objects that are initialized once for the lifetime of a command. Consolidating that to a single package makes it easier to reason about and discover interactions with context

Reply:
I understand that but to me working with context is not a unifying theme to put code together.

BTW, we also have libs/log which also deals with context. Potentially these two will be merged together, they are related.